keps/sig-storage/3476-volume-group-snapshot/README.md

```diff
@@ -0,0 +1,1660 @@
+
+There is also another KEP (https://github.com/kubernetes/enhancements/pull/1051) that introduces APIs to do application snapshot, backup, and restore, but that KEP has a broader scope. In other words, volume group snapshot proposed in this KEP can be used by the application snapshot proposed in the other KEP.
+
+### Goals
+
+* Provide an API to take a snapshot of multiple volumes together.
+
+### Non Goals
+
+* Provide an API to manage multiple volumes together in a group.
+* Provide a generic and extensible group API that may be used to support other features in the future.
+* Provide a VolumeGroup API that supports group replication.
+* Provide a design to facilitate volume placement using the group API.
+
+## Proposal for VolumeGroupSnapshot
+
+This proposal introduces new CRDs VolumeGroupSnapshot, VolumeGroupSnapshotContent, and VolumeGroupSnapshotClass.
+
+### Create VolumeGroupSnapshot
+
+A VolumeGroupSnapshot can be created from multiple PVCs with a label on the PVCs specified by the labelSelector in the VolumeGroupSnapshot if the CSI driver supports the CREATE_DELETE_GET_VOLUME_GROUP_SNAPSHOT capability.
+
+Note: In the following, we will use VolumeGroupSnapshot Controller to refer to the control logic for VolumeGroupSnapshot. This is not a new controller. It will be new control logic added to the existing Snapshot Controller and the csi-snapshotter sidecar.
+
+#### Dynamic provisioning
+
+* Admin creates a VolumeGroupSnapshotClass.
+* User creates a VolumeGroupSnapshot with label selector that matches the label applied to all PVCs to be snapshotted together.
+* This will trigger the VolumeGroupSnapshot controller to create a VolumeGroupSnapshotContent API object, and also call the CreateVolumeGroupSnapshot CSI function.
+* The controller will retrieve all volumeSnapshotHandles in the Volume Group Snapshot from the CSI CreateVolumeGroupSnapshotResponse, create VolumeSnapshotContents pointing to the volumeSnapshotHandles. Then the controller will create VolumeSnapshots pointing to the VolumeSnapshotContents.
+* CreateVolumeGroupSnapshot CSI function response
+  * The CreateVolumeGroupSnapshot CSI function should return a list of snapshots (Snapshot message defined in CSI Spec) in its response. The VolumeGroupSnapshot controller can use the returned list of snapshots to construct corresponding individual VolumeSnapshotContents and VolumeSnapshots, wait for VolumeSnapshots and VolumeSnapshotContents to be bound, and update SnapshotList in the VolumeGroupSnapshot Status and SnapshotContentList in the VolumeGroupSnapshotContent Status.
```

Comment:
Since we are also creating VolumeSnapshot objects which are normally user-defined, can you add a section on naming and how we can handle possible name conflicts with user-created objects?